File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <set>

using namespace std;

int solution(vector<int> elements) {
    int length = 1;
    vector<int> cases;
    
    while (length <= elements.size()) {
        for (int i = 0; i < elements.size(); i++) {
            int sum = 0;
            for (int j = i; j < i + length; j++) {
                sum += elements[j % elements.size()];
            }
            cases.push_back(sum);
        }
        length++;
    }
    
    set<int> s(cases.begin(), cases.end());
    return s.size();
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.py
from collections import deque

def solution(elements):
    cases = set()
    dq = deque(elements)
    length = 1
    
    while length <= len(elements):
        for i in range(len(elements)):
            dq.rotate(1)
            cases.add(sum(list(dq)[:length]))
            
        length += 1
        
    return len(cases)